app/(admin)/admin/articles/page.tsx

```diff
@@ -1,72 +1,33 @@
 'use client';
-import PenIcon from "@/components/icons/symbolic/PenIcon";
-import DropDown from "@/components/shared/DropDown";
-import Table from "@/components/ui/Table/Table";
-import Button from "@/components/shared/Button";
-import { useAppDispatch, useAppSelector } from "@/store/hook";
-import { useEffect } from "react";
-import { allArticles } from "@/store/articles/action";
-
-const articlesHeader = [
-    { id: '1', title: 'Title' },
-    { id: '2', title: 'Status' },
-    { id: '3', title: 'Views' },
-  ];
+import { useEffect, useState } from 'react';
+import { useAppDispatch, useAppSelector } from '@/store/hook';
+import { allArticles } from '@/store/articles/action';
+import ArticlesTable from '@/components/admin/Articles/ArticlesTable';
+import Pagination from '@/components/shared/Pagination';
 
 const ArticlesListPage = () => {
+  const [currentPage, setCurrentPage] = useState(0);
   const getAllArticles = useAppSelector(state => state.articles.articles);
+  const totalPages = useAppSelector(state => state.articles.totalPages);
   const dispatch = useAppDispatch();
 
   useEffect(() => {
-    dispatch(allArticles());
-  }, [dispatch])
-  console.log('allArticles', getAllArticles)
-  return (
-    <>
-      <div className="">
-        {articlesHeader?.map(item => (
-          <div key={item.id}>{item.title}</div>
-        ))}
-        <DropDown
-          icon={<PenIcon />}
-          // classNameBtn=""
-          classNameItem=""
-          label="Add new"
-          items={[
-            { label: 'Article' },
-            { label: 'Event' },
-            { label: 'Program' },
-          ]}
-        />
-      </div>
-
-      <div>
-        {/* <Table
-          classNameRow="bg-admin-100"
-          data={allArticles}
-          renderHeader={() => (
-            <>
-              <th className="pl-[45px] pb-4 border-b border-admin-300">Title</th>
-              <th className="pb-4 border-b  border-admin-300">Status</th>
-              <th className="pb-4 border-b  border-admin-300">Views</th>
-              <th className="pb-4 border-b  border-admin-300 flex justify-end">Actions</th>
-            </>
-          )}
+    dispatch(allArticles({ page: currentPage }));
+  }, [dispatch, currentPage]);
 
-          renderRow={article => {
-            return (
-              <>
-                <td>
+  const changePage = (page: number) => setCurrentPage(page);
 
-                </td>
-              </>
-            )
-          }}
-        /> */}
-
-      </div>
-    </>
+  return (
+    <div className="w-full">
+      <ArticlesTable articles={getAllArticles} />
+
+      <Pagination
+        currentPage={currentPage}
+        totalPages={totalPages}
+        changePage={changePage}
+      />
```

Comment:
Why not place the `Pagination` component inside `ArticlesTable` using the Render Props pattern (so that all `useState` and `useAppSelector` calls are inside the `ArticlesTable` component)? Why do this? First, they are closely related. Second, it prevents a full page re-render when something changes that only concerns the table (since in the future, the page might include more than just the table)

More about Render Props: [https://react.dev/reference/react/cloneElement#passing-data-with-a-render-prop](https://react.dev/reference/react/cloneElement#passing-data-with-a-render-prop)